discord/onboarding.py

```diff
@@ -0,0 +1,322 @@
+copy of this software and associated documentation files (the "Software"),
+to deal in the Software without restriction, including without limitation
+the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the
+Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+"""
+
+from __future__ import annotations
+
+from random import randint
+from typing import TYPE_CHECKING, Any
+
+from .enums import OnboardingMode, PromptType, try_enum
+from .partial_emoji import PartialEmoji
+from .utils import MISSING, _get_as_snowflake, cached_property, get
+
+if TYPE_CHECKING:
+    from .abc import Snowflake
+    from .channel import ForumChannel, TextChannel, VoiceChannel
+    from .emoji import Emoji
+    from .guild import Guild
+    from .object import Object
+    from .partial_emoji import PartialEmoji
+    from .types.onboarding import Onboarding as OnboardingPayload
+    from .types.onboarding import OnboardingPrompt as OnboardingPromptPayload
+    from .types.onboarding import PromptOption as PromptOptionPayload
+
+__all__ = (
+    "Onboarding",
+    "OnboardingPrompt",
+    "PromptOption",
+    "PromptType",
+)
+
+
+class PromptOption:
+    """Represents an onboarding prompt option displayed in :class:`OnboardingPrompt`
+
+    .. versionadded:: 2.5
+
+    Attributes
+    ----------
+
+    id: :class:`int`
+        The id of the prompt option.
+    channels: List[:class:`Snowflake`]
+        The channels assigned to the user when they select this option.
+    roles: List[:class:`Snowflake`]
+        The roles assigned to the user when they select this option.
+    emoji: Union[:class:`Emoji`, :class:`PartialEmoji`]
+        The emoji displayed with the option.
+    title: :class`str`
+        The option's title.
+    description: Optional[:class:`str`]
+        The option's description.
+    """
+
+    def __init__(
+        self,
+        title: str,
+        channels: list[Snowflake] | None = None,
+        roles: list[Snowflake] | None = None,
+        description: str | None = None,
+        emoji: Emoji | PartialEmoji | None = None,
+        id: int | None = None,
+    ):
+        # ID is required when making edits, but it can be any snowflake that isn't already used by another prompt during edits
+        self.id: int | None = id or randint(10000000000000000)

```

Comment:
You could use [`discord.utils.time_snowflake`](https://docs.pycord.dev/en/stable/api/utils.html#discord.utils.time_snowflake) instead of `randint`, right?

Reply:
I did, however if you set multiple prompts it generates the same snowflake which errors on submit.

Reply:
Ah ok